projection/ibc_channel_message/ibc_channel_message.go

```diff
@@ -0,0 +1,316 @@
+				Message:         typedEvent,
+			}
+			messages = append(messages, message)
+
+		} else if typedEvent, ok := event.(*event_usecase.MsgIBCChannelOpenTry); ok {
+
+			channelID := typedEvent.Params.ChannelID
+
+			message := view.IBCChannelMessageRow{
+				ChannelID:       channelID,
+				BlockHeight:     height,
+				BlockTime:       blockTime,
+				TransactionHash: typedEvent.TxHash(),
+				Success:         true,
+				Error:           "",
+				Sender:          "",
+				Receiver:        "",
+				Denom:           "",
+				Amount:          "",
+				MessageType:     typedEvent.MsgName,
+				Message:         typedEvent,
+			}
+			messages = append(messages, message)
+
+		} else if typedEvent, ok := event.(*event_usecase.MsgIBCChannelOpenAck); ok {
+
+			channelID := typedEvent.Params.ChannelID
+
+			message := view.IBCChannelMessageRow{
+				ChannelID:       channelID,
+				BlockHeight:     height,
+				BlockTime:       blockTime,
+				TransactionHash: typedEvent.TxHash(),
+				Success:         true,
+				Error:           "",
+				Sender:          "",
+				Receiver:        "",
+				Denom:           "",
+				Amount:          "",
+				MessageType:     typedEvent.MsgName,
+				Message:         typedEvent,
+			}
+			messages = append(messages, message)
+
+		} else if typedEvent, ok := event.(*event_usecase.MsgIBCChannelOpenConfirm); ok {
+
+			channelID := typedEvent.Params.ChannelID
+
+			message := view.IBCChannelMessageRow{
+				ChannelID:       channelID,
+				BlockHeight:     height,
+				BlockTime:       blockTime,
+				TransactionHash: typedEvent.TxHash(),
+				Success:         true,
+				Error:           "",
+				Sender:          "",
+				Receiver:        "",
+				Denom:           "",
+				Amount:          "",
+				MessageType:     typedEvent.MsgName,
+				Message:         typedEvent,
+			}
+			messages = append(messages, message)
+
+		} else if typedEvent, ok := event.(*event_usecase.MsgIBCTransferTransfer); ok {
+
+			channelID := typedEvent.Params.SourceChannel
+
+			message := view.IBCChannelMessageRow{
+				ChannelID:       channelID,
+				BlockHeight:     height,
+				BlockTime:       blockTime,
+				TransactionHash: typedEvent.TxHash(),
+				Success:         true,
+				Error:           "",
+				Sender:          typedEvent.Params.Sender,
+				Receiver:        typedEvent.Params.Receiver,
+				Denom:           typedEvent.Params.Token.Denom,
+				Amount:          strconv.FormatUint(typedEvent.Params.Token.Amount, 10),
+				MessageType:     typedEvent.MsgName,
+				Message:         typedEvent,
+			}
+			messages = append(messages, message)
+
+		} else if typedEvent, ok := event.(*event_usecase.MsgIBCRecvPacket); ok {
+
+			channelID := typedEvent.Params.Packet.DestinationChannel
+
+			message := view.IBCChannelMessageRow{
+				ChannelID:       channelID,
+				BlockHeight:     height,
+				BlockTime:       blockTime,
+				TransactionHash: typedEvent.TxHash(),
+				Success:         typedEvent.Params.MaybeFungibleTokenPacketData.Success,
```

Comment:
This may break if the packet is not a fungible token packet

Reply:
Good catch! Let me change it.